### PR TITLE
Threads implemented, yaay. 

### PR DESCRIPTION
**Pull request by BorekZnovustvoritel**
Just activated already written code and made some small tweaks. Now pumpkin needs to run in an environment with  the Discord.py 2.0 library installed!

### CODE REVIEW

File: modules/base/base/module.py [hunk 1]
```diff
@@ -188,7 +188,7 @@ async def bookmarks_unset(self, ctx, channel: discord.TextChannel = None):
 
     @commands.guild_only()
     @commands.check(acl.check)
-    @commands.group(enabled=False)
+    @commands.group(enabled=True)
```
Comment (sinus-x) [inline modules/base/base/module.py:191]: The `enabled=True` may be removed completly, it is implicitly set to `True`.

File: modules/base/base/module.py [hunk 2]
```diff
@@ -397,17 +397,56 @@ async def _bookmark(
...
+        """Handle autothread functionality."""
         tc = TranslationContext(payload.guild_id, payload.user_id)
 
-        await utils.Discord.remove_reaction(message, payload.emoji, payload.member)
-        await payload.member.send(tr("_autothread", "wip", tc))
+        for reaction in message.reactions:
+            if reaction.emoji != "🧵":
+                continue
+
+            # remove if the message is threaded or is in channel with threading not enabled
```
Comment (sinus-x) [inline modules/base/base/module.py:411]: The comment says that the reaction is removed when threads are not enabled, but only checks if the message already has a thread.

File: modules/base/base/module.py [hunk 3]
```diff
@@ -397,17 +397,56 @@ async def _bookmark(
...
+            if message.flags.has_thread:
+                await guild_log.debug(
+                    payload.member,
+                    message.channel,
+                    f"Removing {payload.user_id}'s thread: Message has already a thread.",
+                )
+                await reaction.clear()
+                return
+
+            # stop if there isn't enough pins
```
Comment (sinus-x) [inline modules/base/base/module.py:421]: pins => thread reactions

File: modules/base/base/module.py [hunk 4]
```diff
@@ -397,17 +397,56 @@ async def _bookmark(
...
+
+            try:
+                threadName = (
+                    tr("_autothread", "thread", tc) + " " + str(message.author.name)
+                )
+                await message.start_thread(name=threadName)
+                await guild_log.info(
+                    payload.member,
+                    message.channel,
+                    f"Thread opened on a message {message.jump_url}",
```
Comment (sinus-x) [inline modules/base/base/module.py:439]: The log text should contain message author.

File: modules/base/base/module.py [hunk 5]
```diff
@@ -397,17 +397,56 @@ async def _bookmark(
...
+                )
+            except discord.errors.HTTPException:
+                await guild_log.error(
+                    payload.member,
+                    message.channel,
+                    "Could not open a thread on a message.",
```
Comment (sinus-x) [inline modules/base/base/module.py:445]: The log text should contain message link and author.

**Review by sinus-x — CHANGES_REQUESTED** [review]
I left some notes there. I also have a few requests:
- Can you disable this feature if `discord.py` version is 1.x? Simple `return` if the feature is not supported will be enough. And maybe a note in the `autothread get` embed.
- Please, squash your commits into one and use `--amend` to apply new changes, so the final change is only one commit.
- Follow commit message format: Short title under 50 (or 72 at max) characters: `<Module>: <Change description>`, followed by empty line and quick description of the change (e.g. the requirement of pumpkin.py 2.x)

**Review by sinus-x — APPROVED** [review]

**Review by Czechbol — APPROVED** [review]
No objections

**Pull request merged**